### PR TITLE
Gracefully handle anchor creation for non-latin headlines

### LINKED ISSUE #8528: Table of Content ignores first non-roman H2 title
For non-roman translations, especially when the first title doesn't include any non-roman characters and therefore renders the anchor as `/#`, the table of content ignores the header and doesn't render.

In the below screenshot, "미니앱이 아닌 것들" is an H2 header, but it doesn't contain any roman characters and therefore has an anchor of `/#` and doesn't appear on the table of contents.

Screen Shot 2022-08-15 at 5 46 04 PM

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Fixes #8528. Adds a fallback slug method for headlines with non-latin characters.

### CODE REVIEW

**Comment by devnook** [comment]
LGTM, it looks like it might be worthwhile to add a test with some non-latin cases for this plugin though?

File: src/site/_plugins/markdown.js [hunk 1]
```diff
@@ -36,7 +36,20 @@ const markdown = md({
     permalinkClass: 'w-headline-link',
     permalinkSymbol: '#',
     // @ts-ignore
-    slugify: (s) => slugify(s, {lower: true, remove: /[$*_+~.()'"!\-:@]+/g}),
+    slugify: (s) => {
+      let slug = slugify(s, {lower: true, remove: /[$*_+~.()'"!\-:@]+/g});
```
Comment (tunetheweb) [inline src/site/_plugins/markdown.js:40]: @matthiasrohmer is this still needed now #8262 is merged?

If so, and you want to URL encode non-English chars, which I think [is technically the right thing](https://stackoverflow.com/questions/40636281/what-are-the-eligible-characters-in-a-urls-fragment-location-hash), but leads to ugly URLs when they seem to work fine without encoding them, then I think you need to change this as this code no longer works?

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/site/_plugins/markdown.js (hunk 1 above)
Comment (matthiasrohmer) [reply src/site/_plugins/markdown.js:40]: Thanks for flagging here! I'll check and then update accordingly, but probably this is fixed with your change 🙂

Comment (tunetheweb) [reply src/site/_plugins/markdown.js:40]: Good stuff. FYI I’ve a enhancement to that in #8549 as missed question marks in my first PR.

**Comment by matthiasrohmer** [comment]
Superseded by #8263 and #8549.

**Pull request closed without merging**